### ISSUE TITLE
Internal Service Error: Expired Transaction

### ISSUE THREAD

**Issue by JunYong1024** [eoscanada/eos-go#127]
hi, I'm a chinese developer. There is some problem when I call the api **eos.New("xxx").PushTransaction()**, the error info is following:
```
[Internal Service Error: Expired Transaction: expired transaction 8d7b261650f867cf80cb4ff76b5e35dde660154e1a2cebecd865b78d4df3e646].
```
but what really confuse me is that it works in my local debugging enviroment with goland IDE. When I put the code into singapore's server,  it always fail as mentioned above.  
I have googled it but no much userful info for me. Appreciate if you can reslove my trouble:)

**Comment by fproulx-eoscanada** [comment]
Hi.

Can you give us a slightly more complete code snippet.

The default expiration is pretty short and if the latency to EOS Canada is
higher that’s what would happen.

Have you tried pointing to another API node closer to you ?

**Comment by JunYong1024** [comment]
Thanks for your reply.   

Actually, I fixed the problem several hours ago by setting transaction expiration longer...  
`tx.Expiration.Time = tx.Expiration.Add(time.Second * 60)`  

It works but I didn't figure out why. I was new to blockchain and eos:(  

Now I got some answer from your relpy. Here is the API node I used in my code:  
`var eosAPI = eos.New("https://node1.zbeos.com")`  

I don't sure the awkward way I fixed the problem is safe and correct. Maybe I should try some closer node as you suggested.
